controllers/osbuildenvconfig_controller.go

```diff
@@ -490,20 +660,323 @@ func (r *OSBuildEnvConfigReconciler) generateComposerService(serviceName, portNa
 			Ports: []corev1.ServicePort{
 				{
 					Name:       portName,
-					Port:       443,
+					Port:       int32(port),
 					Protocol:   "TCP",
 					TargetPort: intstr.FromInt(targetPort),
 				},
 			},
-			Selector: map[string]string{
-				"app": "osbuild-composer",
-			},
+			Selector: selector,
 		},
 	}
 
 	return service, controllerutil.SetControllerReference(instance, service, r.Scheme)
 }
 
+func (r *OSBuildEnvConfigReconciler) getOrCreateWorkerSSHKeysSecret(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig) (*corev1.Secret, bool, error) {
+	workerSSLSecret, err := r.SecretRepository.Read(ctx, workerSSHKeysSecretName, conf.GlobalConf.WorkingNamespace)
+	if err == nil {
+		return workerSSLSecret, false, nil
+	}
+
+	if errors.IsNotFound(err) {
+		privateKeyBytes, publicKeyBytes, err := sshkey.GenerateSSHKeyPair()
```

Comment:
more logs is needed here

Reply:
Why here? This is a recurring pattern in which the error, if happens, is printed at the top of the calling stack